### ISSUE TITLE
corner case eats up memory and doesn't complete

### ISSUE THREAD

**Issue by jgm** [jgm/highlighting-kate#89]
Originally reported at jgm/pandoc#2977 by @amazing-clemo

To reproduce, `highlighting-kate -s scheme` with the sample below:

```
SW_Etage_2_Sued(config-if-range)#do show port-se
Secure Port MaxSecureAddr CurrentAddr SecurityViolation Security Action
               (Count)       (Count)        (Count)
--------------------------------------------------------------------
        Fa0/1        1          0                 0         Shutdown
        Fa0/2        1          0                 0         Shutdown
        Fa0/3        1          0                 0         Shutdown
        Fa0/4        1          0                 0         Shutdown
        Fa0/5        1          0                 0         Shutdown
        Fa0/6        1          0                 0         Shutdown
        Fa0/7        1          0                 0         Shutdown
        Fa0/8        1          0                 0         Shutdown
        Fa0/9        1          0                 0         Shutdown
       Fa0/10        1          0                 0         Shutdown
       Fa0/11        1          0                 0         Shutdown
       Fa0/12        1          0                 0         Shutdown
       Fa0/13        1          0                 0         Shutdown
       Fa0/14        1          0                 0         Shutdown
       Fa0/15        1          0                 0         Shutdown
       Fa0/16        1          0                 0         Shutdown
       Fa0/17        1          0                 0         Shutdown
       Fa0/18        1          0                 0         Shutdown
       Fa0/19        1          0                 0         Shutdown
----------------------------------------------------------------------
```

**Comment by jgm** [comment]
Simpler repro:
```
echo "#do" | highlighting-kate -s scheme
```

**Comment by amazing-clemo** [comment]
found another sample:
<pre>
```scheme
#delay_warning_time
```
</pre>

and experimented with it. I think it's `#d` followed by something except numbers

I could reproduce it with `#da`, `#de`, `#dA`,  `#d `,  `#dö`,  `#d.` and `#d²`
and could not reproduce it with `#d0`

**Comment by jgm** [comment]
I believe the issue has to do with
``` xml
      <context name="SpecialNumber" attribute="Normal" fallthrough="true" lineEndContext="#pop">
        <RegExpr attribute="Decimal" context="#pop" String="\d*(\.\d+)?"/>
      </context>
```
The regex here matches the empty string.

**Comment by jgm** [comment]
Fixed in 0.6.2.1 release
